src/base/bittorrent/session.cpp

```diff
@@ -5337,6 +5337,7 @@ void Session::handleTrackerAlert(const lt::tracker_alert *a)
     if (!torrent)
         return;
 
+    torrent->m_hasWorkingTracker = torrent->hasWorkingTracker();
```

Comment:
A few lines below there is 
`if (a->type() == lt::tracker_reply_alert::alert_type)` which means that, if i'm reading the docs correctly, announce succeeded i.e. the tracker is working. So you could use the new setter you'll make to set TorrentImpl's `m_hasWorkingTracker` to true in there.